pyproject.toml

```diff
@@ -63,6 +64,7 @@ dependencies = [
     "django>=4.0",
     "requests>=2.4.3",
     "urllib3>=1.25.0",  # requests dependency: fixes RFC 7578 header encoding
+    "typing_extensions>=4.12",  # for older Python compatibility
```

Comment:
I'm happy to implement whatever you request. I have started to rely heavily on Python type checking for my own development, but if you want me to remove all type hints once the work part is done, I'm happy to do that, too.

I confirm your assessment that it has only been added for `TypedDict` and `NotRequired` support in versions <3.11.

> at the very least we could probably make this a version-dependent dependency: "typing_extensions>=4.12; python_version < 3.11"

I will do this for now.

Reply:
I'm not sure what the best practice is for type hints in libs that support multiple python versions, but me intuition tells me there has to be a better way than this(?):

```python3
if sys.version_info < (3, 11):
    from typing_extensions import Any, Dict, List, Literal, NotRequired, TypedDict
else:
    from typing import Any, Dict, List, Literal, NotRequired, TypedDict
```

I suppose if #394 is coming, then maybe it is worth making `typing_extensions` a requirement? That way all of the type imports can come from `typing_extensions` instead of `typing`:

```python3
from typing_extensions import Any, Dict, List, Literal, NotRequired, TypedDict
```

I will do some more research on this.